pennylane/drawer/draw.py

```diff
@@ -260,10 +260,15 @@ def wrapper(*args, **kwargs):
             _wire_order = wire_order or qnode.tape.wires
         else:
             original_expansion_strategy = getattr(qnode, "expansion_strategy", None)
-
             try:
                 qnode.expansion_strategy = expansion_strategy or original_expansion_strategy
                 tapes = qnode.construct(args, kwargs)
+                if isinstance(qnode.device, qml.devices.Device):
+                    if expansion_strategy == "device":
+                        program, _ = qnode.device.preprocess()
+                    else:
+                        program = qnode.transform_program
+                    tapes = program([qnode.tape])

```

Comment:
```suggestion
                    tapes, _ = program([qnode.tape])

```

Reply:
I'm confused. Doesn't calling a transform program return tapes and a processing function?

Reply:
Yes I was too, but you will see that later it loops over `tapes[0]`, we can fix it when the draw is reworked.